experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/unboxed.spec.ts

```diff
@@ -373,35 +689,108 @@ describe("unboxed unit tests", () => {
 				assert.equal(bar.value, true);
 			});
 
-			// it("Recursive", () => {
-			// 	const builder = new SchemaBuilder("test");
-			// 	const stringLeafSchema = builder.leaf("string", ValueSchema.String);
-			// 	const booleanLeafSchema = builder.leaf("boolean", ValueSchema.Boolean);
-			// 	const mapSchema = builder.mapRecursive(
-			// 		"map",
-			// 		SchemaBuilder.fieldRecursive(FieldKinds.optional, [
-			// 			stringLeafSchema,
-			// 			booleanLeafSchema,
-			// 			() => mapSchema,
-			// 		]),
-			// 	);
-			// 	const rootSchema = SchemaBuilder.fieldOptional(mapSchema);
-			// 	const schema = builder.intoDocumentSchema(rootSchema);
-
-			// 	const { context, cursor } = initializeTreeWithContent(schema, {
-			// 		foo: "Hello world",
-			// 		bar: true,
-			// 	});
-			// cursor.firstNode(); // Root node field has 1 node; move into it
-
-			// 	const unboxed = unboxedTree(context, mapSchema, cursor);
-			// 	assert.equal(unboxed.size, 2);
-			// 	assert.equal(unboxed.get(brand("foo")), "Hello world");
-			// 	assert.equal(unboxed.get(brand("bar")), true);
-			// });
+			it("Recursive", () => {
+				const builder = new SchemaBuilder("test");
+				const stringLeafSchema = builder.leaf("string", ValueSchema.String);
+				const booleanLeafSchema = builder.leaf("boolean", ValueSchema.Boolean);
+				const mapSchema = builder.mapRecursive(
+					"map",
+					SchemaBuilder.fieldRecursive(
+						FieldKinds.optional,
+						stringLeafSchema,
+						booleanLeafSchema,
+						() => mapSchema,
+					),
+				);
+				const rootSchema = SchemaBuilder.fieldOptional(mapSchema);
+				const schema = builder.intoDocumentSchema(rootSchema);
+
+				const { context, cursor } = initializeTreeWithContent(schema, {
+					foo: "Hello world",
+					bar: true,
+				});
+				cursor.firstNode(); // Root node field has 1 node; move into it
+
+				const unboxed = unboxedTree(context, mapSchema, cursor);
+				assert.equal(unboxed.size, 2);
+
+				// Map value type is not known, so nodes will not be unboxed.
+				const fooEntry = unboxed.get(brand("foo"));
+				assert(fooEntry !== undefined);
+				assert.equal(fooEntry.type, "string");
+				assert.equal(fooEntry.value, "Hello world");
+
+				const barEntry = unboxed.get(brand("bar"));
+				assert(barEntry !== undefined);
+				assert.equal(barEntry.type, "boolean");
+				assert.equal(barEntry.value, true);
+			});
 		});
 	});
 	describe("unboxedUnion", () => {
-		// TODO
+		describe("Value field", () => {
```

Comment:
the cases of unions that are currently interesting are Any, 1 thing and more than one thing. I'd expect one test for each of these, and that would be it. Field kinds should not be involved here, there is seperate testing for that above.